--- zabbix-1.6.4/frontends/php/include/config.inc.php.orig	2009-04-09 10:26:23.000000000 +0200
+++ zabbix-1.6.4/frontends/php/include/config.inc.php	2009-04-09 10:29:00.000000000 +0200
@@ -101,8 +101,7 @@ function TODO($msg) { echo "TODO: ".$msg
 	$ZBX_LOCALNODEID = 0;
 	$ZBX_LOCMASTERID = 0;
 
-	$ZBX_CONFIGURATION_FILE = './conf/zabbix.conf.php';
-	$ZBX_CONFIGURATION_FILE = realpath(dirname($ZBX_CONFIGURATION_FILE)).'/'.basename($ZBX_CONFIGURATION_FILE);
+	$ZBX_CONFIGURATION_FILE = '/etc/zabbix/web/zabbix.conf.php';
 
 	unset($show_setup);
 
