diff -up zabbix-1.8.2/frontends/php/include/config.inc.php.orig zabbix-1.8.2/frontends/php/include/config.inc.php
--- zabbix-1.8.2/frontends/php/include/config.inc.php.orig	2010-03-29 19:22:45.000000000 +0200
+++ zabbix-1.8.2/frontends/php/include/config.inc.php	2010-03-30 11:11:41.000000000 +0200
@@ -110,8 +110,7 @@ function __autoload($class_name){
 	$ZBX_LOCALNODEID = 0;
 	$ZBX_LOCMASTERID = 0;
 
-	$ZBX_CONFIGURATION_FILE = './conf/zabbix.conf.php';
-	$ZBX_CONFIGURATION_FILE = realpath(dirname($ZBX_CONFIGURATION_FILE)).'/'.basename($ZBX_CONFIGURATION_FILE);
+	$ZBX_CONFIGURATION_FILE = '/etc/zabbix/web/zabbix.conf.php';
 
 	unset($show_setup);
 
