diff -up zabbix-1.8.8/frontends/php/include/screens.inc.php.no-flash zabbix-1.8.8/frontends/php/include/screens.inc.php
--- zabbix-1.8.8/frontends/php/include/screens.inc.php.no-flash	2011-10-05 11:26:07.000000000 +0200
+++ zabbix-1.8.8/frontends/php/include/screens.inc.php	2011-10-05 11:26:30.000000000 +0200
@@ -380,7 +380,6 @@ require_once('include/js.inc.php');
 		$form->addVar('screenid',$_REQUEST['screenid']);
 // a-z order!!!
 		$cmbRes = new CCombobox('resourcetype',$resourcetype,'submit()');
-		$cmbRes->addItem(SCREEN_RESOURCE_CLOCK,				S_CLOCK);
 		$cmbRes->addItem(SCREEN_RESOURCE_DATA_OVERVIEW,		S_DATA_OVERVIEW);
 		$cmbRes->addItem(SCREEN_RESOURCE_GRAPH,				S_GRAPH);
 		$cmbRes->addItem(SCREEN_RESOURCE_ACTIONS,			S_HISTORY_OF_ACTIONS);
