diff -Nur zabbix-2.2.3/src/zabbix_agent/logfiles.c zabbix-2.2.3-zbx8238/src/zabbix_agent/logfiles.c
--- zabbix-2.2.3/src/zabbix_agent/logfiles.c	2014-04-07 11:03:41.000000000 +0200
+++ zabbix-2.2.3-zbx8238/src/zabbix_agent/logfiles.c	2014-06-08 23:13:38.291781722 +0200
@@ -930,6 +930,9 @@
 		goto out;
 	}
 
+	if (NULL != mtime)
+		*mtime = (int)buf.st_mtime;
+
 	if ((zbx_uint64_t)buf.st_size == *lastlogsize)
 	{
 		/* The file size has not changed. Nothing to do. Here we do not deal with a case of changing */
@@ -961,9 +964,6 @@
 		*lastlogsize = l_size;
 		*skip_old_data = 0;
 
-		if (NULL != mtime)
-			*mtime = (int)buf.st_mtime;
-
 		ret = zbx_read2(f, lastlogsize, mtime, big_rec, encoding, regexps, pattern, output_template, p_count,
 				s_count, process_value, server, port, hostname, key);
 	}
