diff -Nur zabbix-2.0.3/frontends/php/include/defines.inc.php zabbix-2.0.3-fonts/frontends/php/include/defines.inc.php
--- zabbix-2.0.3/frontends/php/include/defines.inc.php	2012-10-03 16:41:36.000000000 +0200
+++ zabbix-2.0.3-fonts/frontends/php/include/defines.inc.php	2012-10-05 23:47:17.341790748 +0200
@@ -34,7 +34,7 @@
 
 define('ZBX_WIDGET_ROWS', 20);
 
-define('ZBX_FONTPATH',				realpath('fonts')); // where to search for font (GD > 2.0.18)
+define('ZBX_FONTPATH',				'/usr/share/fonts/dejavu'); // where to search for font (GD > 2.0.18)
 define('ZBX_GRAPH_FONT_NAME',		'DejaVuSans'); // font file name
 define('ZBX_GRAPH_LEGEND_HEIGHT',	120); // when graph height is less then this value, some legend will not show up
 
