Index: frontends/php/api/classes/CService.php
===================================================================
--- frontends/php/api/classes/CService.php	(.../2.0)	(revision 38431)
+++ frontends/php/api/classes/CService.php	(.../dev/ZBX-6992)	(revision 38466)
@@ -1004,8 +1004,8 @@
 		// since multiple alarms can have the same timestamp, we only need to save the last one
 		$query = DBSelect(
 			'SELECT sa.serviceid,sa.value'.
-			' FROM (SELECT MAX(sa3.servicealarmid) servicealarmid'.
-					' FROM (SELECT sa2.serviceid,MAX(sa2.clock) clock'.
+			' FROM (SELECT MAX(sa3.servicealarmid) AS servicealarmid'.
+					' FROM (SELECT sa2.serviceid,MAX(sa2.clock) AS clock'.
 							' FROM service_alarms sa2'.
 							' WHERE sa2.clock<'.zbx_dbstr($beforeTime).
 								' AND '.dbConditionInt('sa2.serviceid', $serviceIds).
