diff -up zabbix-2.0.1/frontends/php/include/defines.inc.php.orig zabbix-2.0.1/frontends/php/include/defines.inc.php
--- zabbix-2.0.1/frontends/php/include/defines.inc.php.orig	2012-06-27 23:03:03.000000000 +0200
+++ zabbix-2.0.1/frontends/php/include/defines.inc.php	2012-07-12 16:19:11.000000000 +0200
@@ -31,7 +31,7 @@ define('ZBX_PERIOD_DEFAULT',			3600); //
 
 define('ZBX_WIDGET_ROWS',				20);
 
-define('ZBX_FONTPATH',					realpath('fonts')); // where to search for font (GD > 2.0.18)
+define('ZBX_FONTPATH',					'/usr/share/fonts/dejavu'); // where to search for font (GD > 2.0.18)
 define('ZBX_GRAPH_FONT_NAME',			'DejaVuSans'); // font file name
 define('ZBX_GRAPH_LEGEND_HEIGHT',		120); // when graph height is less then this value, some legend will not show up
 
