commit 8332467b83652ba5b82d9c13732f35b4e5ea19c7
Author: Volker Fröhlich <volker27@gmx.at>
Date:   Thu Sep 11 10:55:31 2014 +0200

    Remove Flash clock for screens

diff --git a/frontends/php/include/screens.inc.php b/frontends/php/include/screens.inc.php
index f3ca635..52c4c33 100644
--- a/frontends/php/include/screens.inc.php
+++ b/frontends/php/include/screens.inc.php
@@ -25,7 +25,6 @@ require_once dirname(__FILE__).'/js.inc.php';
 
 function screen_resources($resource = null) {
 	$resources = array(
-		SCREEN_RESOURCE_CLOCK => _('Clock'),
 		SCREEN_RESOURCE_DATA_OVERVIEW => _('Data overview'),
 		SCREEN_RESOURCE_GRAPH => _('Graph'),
 		SCREEN_RESOURCE_ACTIONS => _('Action log'),
